language-server/test/dotty/tools/languageserver/CompletionTest.scala

```diff
@@ -1130,4 +1130,98 @@ class CompletionTest {
     code"""import scala.util.chaining.`s${m1}"""
              .withSource.completion(m1, expected)
   }
+
+  @Test def matchTypeCompletions: Unit = {
+    val expected = Set(
+      ("fooTest", Method, "(y: Int): Int"),
+    )
+    code"""case class Foo(x: Int) {
+           |  def fooTest(y: Int): Int = ???
+           |}
+           |type Elem[X] = X match {
+           |  case Int => Foo
+           |  case Any => X
+           |}
+           |def elem[X](x: X): Elem[X] = x match {
+           |  case x: Int => Foo(x)
+           |  case x: Any => x
+           |}
+           |object Test: 
+           |  elem(1).foo${m1}"""
+             .withSource.completion(m1, expected)
+  }
+
+  @Test def higherKindedMatchTypeDeclaredCompletion: Unit = {
+    val expected = Set(
+      ("map", Method, "[B](f: Int => B): Foo[B]"),
+    )
+    code"""trait Foo[A] {
+           |  def map[B](f: A => B): Foo[B] = ???
+           |}
+           |case class Bar[F[_]](bar: F[Int])
+           |type M[T] = T match {
+           |  case Int => Foo[Int]
+           |}
+           |object Test:
+           |  val x = Bar[M](new Foo[Int]{})
+           |  x.bar.m${m1}"""
+             .withSource.completion(m1, expected)
+  }
+
+  @Test def higherKindedMatchTypeLazyCompletion: Unit = {
```

Comment:
What's the essential difference between this test case and `higherKindedMatchTypeDeclaredCompletion`?

Reply:
higherKindedMatchTypeDeclaredCompletion is reduced and was working properly without this fix, added this one as a regression test.